### DISCUSSION TITLE
Using termux-share with && in bash

### DISCUSSION THREAD

**Discussion by tathastu871** [termux/termux-app#3055] [Ideas]
my script produces temporary images.
I want to view them and delete them after vieving
eg. termux-share file.jpg && rm file.jpg
but even before intent is shared and viewed it deletes the file

Maybe add --delete-after flag to termux-share to delete file after viewing
@agnostic-apollo 
@Grimler91

**Comment by tathastu871** [comment]
https://stackoverflow.com/questions/4609129/delete-file-after-sharing-via-intent

**Comment by agnostic-apollo** [comment]
`termux-share` api does not wait after sharing currently and only calls `startActivity()` instead of `startActivityForResult()`, and `ACTION_SEND` is not supposed to return a result as per docs anyways, although it should wait, so may be useful in some cases.

https://stackoverflow.com/questions/16622012/startactivityforresults-always-returns-result-cancelled-for-intent-action-send

But one can't be sure if target app is actually done with the file or not even if its receiver activity is finished. You can use `termux-share file.jpg && read -n 1 && rm file.jpg` to put the terminal in waiting state until a key is pressed before the file is deleted.

**Comment by agnostic-apollo** [comment]
`termux-share` api does not wait after sharing currently and only calls `startActivity()` instead of `startActivityForResult()`, and `ACTION_SEND` is not supposed to return a result as per docs anyways, although it should wait, so may be useful in some cases.

https://stackoverflow.com/questions/16622012/startactivityforresults-always-returns-result-cancelled-for-intent-action-send

But one can't be sure if target app is actually done with the file or not even if its receiver activity is finished. You can use `termux-share file.jpg && read -n 1 && rm file.jpg` to put the terminal in waiting state until a key is pressed before the file is deleted.
